pkg/rng/entropy.go

```diff
@@ -0,0 +1,160 @@
+package rng
+
+import (
+	"errors"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/insomniacslk/systemboot/pkg/recovery"
+)
+
+// The concept
+//
+// Most systems in real world application do not provide enough entropy at boot
+// time. Therefore we will seed /dev/random with /dev/hwrng if a HW random
+// number generator is available. Entropy is important for cryptographic
+// protocols running in network stacks. Also disk encryption can be a problem
+// if bad or no entropy is available. It can either block provisioning or makes
+// a symmetric key easy to re-calculate.
+
+var (
+	// HwRandomCurrentFile shows/sets the current
+	// HW random number generator
+	HwRandomCurrentFile = "/sys/class/misc/hw_random/rng_current"
+	// HwRandomAvailableFile shows the current available
+	// HW random number generator
+	HwRandomAvailableFile = "/sys/class/misc/hw_random/rng_available"
+	// RandomEntropyAvailableFile shows how much of the entropy poolsize is used
+	RandomEntropyAvailableFile = "/proc/sys/kernel/random/entropy_avail"
+	// EntropyFeedTime sets the loop time for seeding /dev/random by /dev/hwrng
+	// in seconds
+	EntropyFeedTime time.Duration = 2
```

Comment:
this should be `EntropyFeedTime = 2 * time.Second`. time.Duration is expressed in nanoseconds, so that timeout means 2 nanoseconds

Reply:
Done